pkg/kube/kubectl.go

```diff
@@ -102,18 +103,29 @@ func (k *KubectlOperation) Create(spec io.Reader, namespace string) (*crv1alpha1
 	return objRef, err
 }
 
-// Delete k8s resource referred by objectReference
-func (k *KubectlOperation) Delete(objRef crv1alpha1.ObjectReference, namespace string) (*crv1alpha1.ObjectReference, error) {
+// Delete k8s resource referred by objectReference. Waits for the resource to be deleted
+func (k *KubectlOperation) Delete(ctx context.Context, objRef crv1alpha1.ObjectReference, namespace string) (*crv1alpha1.ObjectReference, error) {
 	if namespace == "" {
 		namespace = metav1.NamespaceDefault
 	}
 	if objRef.Namespace != "" {
 		namespace = objRef.Namespace
 	}
-	err := k.dynCli.Resource(schema.GroupVersionResource{Group: objRef.Group, Version: objRef.APIVersion, Resource: objRef.Resource}).Namespace(namespace).Delete(context.Background(), objRef.Name, metav1.DeleteOptions{})
-	if apierrors.IsNotFound(err) {
-		return &objRef, nil
+	err := k.dynCli.Resource(schema.GroupVersionResource{Group: objRef.Group, Version: objRef.APIVersion, Resource: objRef.Resource}).Namespace(namespace).Delete(ctx, objRef.Name, metav1.DeleteOptions{})
+	if err != nil {
+		return &objRef, err
 	}
+	return waitForResourceDeletion(ctx, k, objRef, namespace)
+}
 
+// Wait for resource to get deleted by repeatedly checking for NotFound error
+func waitForResourceDeletion(ctx context.Context, k *KubectlOperation, objRef crv1alpha1.ObjectReference, namespace string) (*crv1alpha1.ObjectReference, error) {
```

Comment:
```suggestion
// waitForResourceDeletion waits for resource to get deleted by repeatedly checking for NotFound error
func waitForResourceDeletion(ctx context.Context, k *KubectlOperation, objRef crv1alpha1.ObjectReference, namespace string) (*crv1alpha1.ObjectReference, error) {
```
General golang convention is to start the comment with the function name we are describing